components/dbutil/src/main/java/com/wso2telco/core/dbutils/AxiataDbService.java

```diff
@@ -1771,4 +1775,107 @@ public boolean insertSmsRequestIds(String requestID, String senderAddress, Map<S
 
         return spToken;
     }
+
+    public SpendLimitDAO getGroupTotalDayAmount(String groupName, String operator, String msisdn)
+            throws AxataDBUtilException {
+
+        Connection con = null;
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        SpendLimitDAO spendLimitDAO = null;
+
+        try {
+            con = DbUtils.getAxiataDBConnection();
+
+            Calendar calendarFrom = Calendar.getInstance();
+            calendarFrom.set(Calendar.AM_PM, Calendar.AM);
+            calendarFrom.set(Calendar.HOUR, 00);
+            calendarFrom.set(Calendar.MINUTE, 00);
+            calendarFrom.set(Calendar.SECOND, 00);
+            calendarFrom.set(Calendar.MILLISECOND, 00);
+
+            Calendar calendarTo = Calendar.getInstance();
+            calendarTo.set(Calendar.AM_PM, Calendar.PM);
+            calendarTo.set(Calendar.HOUR, 11);
+            calendarTo.set(Calendar.MINUTE, 59);
+            calendarTo.set(Calendar.SECOND, 59);
+            calendarTo.set(Calendar.MILLISECOND, 999);
+
+            String sql = "SELECT SUM(amount) AS amount FROM spendlimitdata  where effectiveTime between ? and ? and "
```

Comment:
You are write it convert into string builder but i would be more nicer and clean explicit string builder so ,convert them into string builder. 
eg:
x.append("SELECT SUM(amount) ,age");
x.append(" FROM ").append(" table1  as t1)
x.append("Table2 as t2 ");
x.append(" where t1.id= t2.id");

likewise we can separate the sql into logical block that leads to more maintainable